iso.bbx

```diff
@@ -114,87 +135,181 @@
 \DeclareFieldFormat{doi}{%
-  \printtext{DOI}\addcolon\space
+  \mkbibacro{DOI}\addcolon\space
   \ifhyperref
     {\href{https://doi.org/#1}{\nolinkurl{#1}}}
     {\nolinkurl{#1}}%
 }
+
+% Format medium type in square brackets
 \DeclareFieldFormat{howpublished}{\mkbibbrackets{#1}}
+% If it is online entry type, defaults to [online]
 \DeclareFieldFormat[online]{howpublished}{%
   \iffieldundef{howpublished}
     {\mkbibbrackets{online}}
     {\mkbibbrackets{#1}}%
 }
 
+% Format type, e.g. type of thesis
+% If available, use the localisation string for the key
 \DeclareFieldFormat{type}{%
   \ifbibstring{#1}
     {\mainlstring{#1}}
     {#1}%
 }
 
+% Format supervisor of thesis (precedes the name with a localisation string)
 \DeclareFieldFormat{supervisor}{%
   \mainlstring{bysupervisor}\addspace#1}
 
-%We must handle more than one isbn and issn
+% We must handle more than one isbn and issn
 \DeclareFieldFormat{isbn}{%
-  %\renewcommand*{\do}[1]{\printtext{ISBN}\addspace#1\adddot\addspace}%
-  \printtext{ISBN}\addspace#1}%
+  %\renewcommand*{\do}[1]{\mkbibacro{ISBN}\addspace#1\adddot\space}%
+  \mkbibacro{ISBN}\addspace#1}%
 
 \DeclareFieldFormat{issn}{%
-  \renewcommand*{\do}[1]{\printtext{ISSN}\addspace#1\adddot\addspace}%
+  \renewcommand*{\do}[1]{\mkbibacro{ISSN}\addspace#1\adddot\space}%
   \docsvfield{issn}}%
 
-\DeclareFieldFormat{isan}{\printtext{ISAN}\addspace#1}
-\DeclareFieldFormat{ismn}{\printtext{ISMN}\addspace#1}
-\DeclareFieldFormat{isrn}{\printtext{ISRN}\addspace#1}
-\DeclareFieldFormat{iswc}{\printtext{ISWC}\addspace#1}
+% Format other standard identifiers, precedes the value with the name
+% of identifier
+\DeclareFieldFormat{isan}{\mkbibacro{ISAN}\addspace#1}
+\DeclareFieldFormat{ismn}{\mkbibacro{ISMN}\addspace#1}
+\DeclareFieldFormat{isrn}{\mkbibacro{ISRN}\addspace#1}
+\DeclareFieldFormat{iswc}{\mkbibacro{ISWC}\addspace#1}
 
+% Format url seen date with preceding localisation string in square brackets
 \DeclareFieldFormat{urldate}{\mkbibbrackets{\mainsstring{urlseen}\space#1}}
 
+% a)
+% Format circa date with preceding localisation string in square brackets;
+% default \datecircaprint is \bibstring{circa}\printdelim{datecircadelim},
+% e.g. [ca. 1940].
+% b)
+% Adds non-breaking space after open date range, e.g. '1492-- '.
+% c)
+% If there is no date field in the bib entry,
+% we should state there is no date by printing
+% a localisation string, e.g. [n.d.].
+\DeclareFieldFormat{date}{%
+  % Check any end year for open date range
+  \ifboolexpr{
+    test {\iffieldequalstr{endyear}{}}
+    or test {\iffieldequalstr{eventendyear}{}}
+    or test {\iffieldequalstr{origendyear}{}}
+  }
+    {\ifdatecirca
+      {\mkbibbrackets{#1\mbox{\addnbspace}}}%
+      {#1\mbox{\addnbspace}}}%
+    {\ifdatecirca
+      {\mkbibbrackets{#1}}%
+      {% Check no year is available
+        \ifboolexpr{
+          test {\iffieldundef{year}}
+          and test {\iffieldundef{eventyear}}
+          and test {\iffieldundef{origyear}}
+        }
```

Comment:
Since the `.bbl` contains
```latex
\field{labeldatesource}{nodate}
```
in such a `nodate` situation, it should be possible to check for it with
```latex
\iffieldequalstr{labeldatesource}{nodate}
```
(I haven't tested that, sorry).

It should be noted that in any case this test only makes sense for `labeldate` and not for `date`.